### PR TITLE
Improve performance for data loading and implement new interface 

### PR DESCRIPTION
**Pull request by arthuridea**
New: Implements of async interface
Update: Call AsNoTracking() method for readonly data access
TODO:  Add async UnitTest when Casbin.Net nuget package update.

#24

### CODE REVIEW

File: EFCore-Adapter/CasbinDbAdapter.cs [hunk 1]
```diff
@@ -197,36 +201,67 @@ private CasbinRule<TKey> SavePolicyLine(string ptype, IList<string> rule)
...
+            }
+            if (fieldIndex <= 5 && 5 < fieldIndex + len)
+            {
+                line.V5 = fieldValues[5 - fieldIndex];
+            }
+            return line;
+        }
+
+        private void applyQueryFilter(ref IQueryable<CasbinRule<TKey>> query, CasbinRule<TKey> line)
+        {
```
Comment (Sagilio) [inline EFCore-Adapter/CasbinDbAdapter.cs:239]: I think the `ref` have poorly maintainable, we can remove it and use it like this :
```csharp
private IQueryable<CasbinRule<TKey>> ApplyQueryFilter(IQueryable<CasbinRule<TKey>> query, CasbinRule<TKey> line)
{
    ... 
    return query;
}
```
What do you think?

Comment (arthuridea) [reply EFCore-Adapter/CasbinDbAdapter.cs:239]: I think the keyword `ref` is just the case when we benifit on a side effect of parameter modification. Removing the `ref` keyword,we need to call this method in a way ` quey = applyQueryFilter(query, line); ` which seems to be ugly ( just in my opinion _(:зゝ∠)_ 

but both ways are acceptable.

Comment (arthuridea) [reply EFCore-Adapter/CasbinDbAdapter.cs:239]: But change the method name to "ApplyQueryFilter" is more meaningful. I's my mistake that I break the typo style with other methods.

Comment (Sagilio) [reply EFCore-Adapter/CasbinDbAdapter.cs:239]: I just want to communicate about it and I also think both ways are acceptable. Change the method name to "ApplyQueryFilter" is more meaningful than that.

Comment (arthuridea) [reply EFCore-Adapter/CasbinDbAdapter.cs:239]: > I just want to communicate about it and I also think both ways are acceptable. Change the method name to "ApplyQueryFilter" is more meaningful than that.

OK.

**Comment by hsluoyz** [comment]
@nodece please review.

**Pull request merged**